bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy

```diff
@@ -1104,11 +1104,19 @@ class ApiController {
     Meeting meeting = ServiceUtils.findMeetingFromMeetingID(params.meetingID);
 
     if (meeting != null){
-      uploadDocuments(meeting, true);
-      withFormat {
-        xml {
-          render(text: responseBuilder.buildInsertDocumentResponse("Presentation is being uploaded", RESP_CODE_SUCCESS)
-                  , contentType: "text/xml")
+      if (uploadDocuments(meeting, true)) {
+        withFormat {
+          xml {
+            render(text: responseBuilder.buildInsertDocumentResponse("Presentation is being uploaded", RESP_CODE_SUCCESS)
+                    , contentType: "text/xml")
+          }
+        }
+      } else {
+        withFormat {
+          xml {
+            render(text: responseBuilder.buildInsertDocumentResponse("Presentation feature is disabled, ignoring.",
+                    RESP_CODE_FAILED), contentType: "text/xml")
+          }
         }
       }
```

Comment:
It's a little odd to assume automatically that if `uploadDocuments` failed it is because "Presentation feature is disabled".

I rather suggest to make some validation here as you made in `PresentationController`:
```
if (meetingService.isMeetingWithDisabledPresentation(meetingId)) {
```

This way it will be easier to maintain if in the feature we add more validations within `uploadDocuments`.